src/server/qgsconfigcache.h

```diff
@@ -65,20 +66,46 @@ class SERVER_EXPORT QgsConfigCache : public QObject
      */
     const QgsProject *project( const QString &path, const QgsServerSettings *settings = nullptr );
 
+    /**
+     * Set the invalidation check interval
+     *
+     * If \a msec is set to a positive value then the cache will check for invalidation
+     * every \a msec milliseconds
+     *
+     * If \a msec is set to 0 (or negative value), which is the default, (or negative value) then the cache will use the QFileSystemWatcher for
+     * invalidating cache
+     *
+     * \since QGIS 3.24
+     */
+    void setCheckInterval( int msec );
+
   private:
     QgsConfigCache() SIP_FORCE;
 
-    //! Check for configuration file updates (remove entry from cache if file changes)
-    QFileSystemWatcher mFileSystemWatcher;
-
     //! Returns xml document for project file / sld or 0 in case of errors
     QDomDocument *xmlDocument( const QString &filePath );
 
     QCache<QString, QDomDocument> mXmlDocumentCache;
-    QCache<QString, QgsProject> mProjectCache;
+    QCache<QString, std::pair<QDateTime, std::unique_ptr<QgsProject> > > mProjectCache;
```

Comment:
AMHA this would be confusing: there is only one cache, this is not related specifically to the timer cache: it is a replacement for 'QCache<project>' that add the lastModified Time  QDateTime.

Reply:
Fair point.